sunpy/data/_sample.py

```diff
@@ -83,7 +83,11 @@ def download_sample_data(progress=True, overwrite=True):
         for base_url in _base_urls:
             full_file_name = file_name[0] + file_name[1]
             if url_exists(os.path.join(base_url, full_file_name)):
-                f = download_file(os.path.join(base_url, full_file_name))
+                try:
+                    f = download_file(os.path.join(base_url, full_file_name))
+                except Exception as e:
```

Comment:
I am not sure if the catch all here is a good idea or not. If the download fails, we want to retry, but what if it fails because the disk is full or some other non-mirror related error?

Opionions @derdon @dpshelio?